handler/rfc8628/strategy_hmacsha.go

```diff
@@ -0,0 +1,84 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package rfc8628
+
+import (
+	"context"
+
+	"github.com/ory/x/randx"
+	"github.com/patrickmn/go-cache"
+	"golang.org/x/time/rate"
+
+	"github.com/ory/fosite"
+	enigma "github.com/ory/fosite/token/hmac"
+)
+
+type DefaultDeviceStrategy struct {
+	Enigma           *enigma.HMACStrategy
+	RateLimiterCache *cache.Cache
+	Config           interface {
```

Comment:
one config that implements two interfaces? couldn't we have two separate fields?